templates/init/functions/typescript/_gitignore

```diff
@@ -4,3 +4,7 @@
 
 # Typescript v1 declaration files
 typings/
+
+lib/
```

Comment:
If I create a file structure:

```
project/
  functions/
    lib/
      [built files]
    src/
      index.ts
      lib/
        foo.ts
```

`src/lib/foo.ts` is ignored because of this rule (and so does `lib/`, which is intended). just saying `lib/` is a little too open (though I get that we want to ignore built files).

That said, there are two options:
- assuming the `lib` directory will _always_ contain built `js` and `js.map` files (as ignored above), we should remove `lib` from `.gitignore`. No files other than those two extensions means that `git` will ignore the directory
- if we _want_ to allow `lib` folders elsewhere (and not cause headaches for folks, probably a good idea), we should include the line `!src/**/lib/` which tells it to _allow_ any folder named lib as a sub directory of a folder named `src`

Thoughts?

Reply:
Just FYI, if you use either `/lib/` or `lib/**/*` it will only match from the root (or rather, from the location of the `.gitingore` file) which is probably what you want.

Reply:
Wow. Yup. I struggled to get that right. Thank you @jsayol

Reply:
Good catch - I'm going to remove the line, since it doesn't look like the Typescript compiler will put any files other than .js or js.map files into the output directory. If there is something else in there that wouldn't be matched by the first 2 rules, it was probably added by the developer for some reason, and ignoring it by default is probably not the expected behavior. Also, if the user changes the output directory for some reason, the /lib/ rule will be extraneous.